src/hypothesis/strategies.py

```diff
@@ -197,25 +197,18 @@ def integers(min_value=None, max_value=None):
     max_value is not None then all values will be <= max_value
 
     """
-
     check_valid_bound(min_value, 'min_value')
     check_valid_bound(max_value, 'max_value')
     check_valid_interval(min_value, max_value, 'min_value', 'max_value')
 
-    from hypothesis.searchstrategy.numbers import IntegersFromStrategy, \
-        BoundedIntStrategy, WideRangeIntStrategy
+    min_int_value = None if min_value is None else ceil(min_value)
+    max_int_value = None if max_value is None else floor(max_value)
 
-    min_int_value = None
-    if min_value is not None:
-        min_int_value = int(min_value)
-        if min_int_value != min_value and min_value > 0:
-            min_int_value += 1
+    check_valid_interval(min_int_value, max_int_value,
+                         'ceil(min_value)', 'floor(max_value)')
```

Comment:
I feel like it might be better to give a more explicit error message here like "No integers between min_value=%r and max_value=%r"